From edcae83af3670289daff653fc768063993804115 Mon Sep 17 00:00:00 2001
From: Danct12 <danct12@disroot.org>
Date: Fri, 10 Dec 2021 23:01:34 +0700
Subject: [PATCH 2/3] arm64: dts: rk3399-pinephone-pro: Remove modem node

Since we don't use modem-power driver, this can be removed
for eg25-manager.
---
 .../dts/rockchip/rk3399-pinephone-pro.dts     | 40 +------------------
 1 file changed, 2 insertions(+), 38 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 41738a73a..a63aa9d80 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -407,6 +407,7 @@ vcc_4g_5v: vcc-4g-5v {
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc5v0_sys>;
+		regulator-always-on;
 	};
 
 	vcc_4g: vcc-4g {
@@ -419,6 +420,7 @@ vcc_4g: vcc-4g {
 		regulator-min-microvolt = <3800000>;
 		regulator-max-microvolt = <3800000>;
 		vin-supply = <&vcc_sys>;
+		regulator-always-on;
 	};
 
 	// XXX: modem codec supplies:
@@ -1251,31 +1253,6 @@ mipi_in_panel: endpoint {
 
 &uart3 {
 	status = "okay";
-
-	modem {
-		compatible = "quectel,eg25";
-		char-device-name = "modem-power";
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&modem_control_pins>;
-
-		power-supply = <&vcc_4g>;
-		vbus-supply = <&vcc_4g_5v>;
-
-		enable-gpios = <&gpio0 RK_PB0 GPIO_ACTIVE_HIGH>; // W_DISABLE#
-		reset-gpios = <&gpio3 RK_PB0 GPIO_ACTIVE_HIGH>;
-		status-gpios = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
-		pwrkey-gpios = <&gpio0 RK_PB5 GPIO_ACTIVE_HIGH>;
-
-		host-ready-gpios = <&gpio0 RK_PB4 GPIO_ACTIVE_HIGH>; // apready
-		wakeup-gpios = <&gpio0 RK_PA1 GPIO_ACTIVE_HIGH>; // ri
-
-		dtr-gpios = <&gpio0 RK_PA3 GPIO_ACTIVE_HIGH>;
-		cts-gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_HIGH>;
-		rts-gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_HIGH>;
-
-		quectel,qdai = "3,0,0,4,0,1,1,1";
-	};
 };
 
 &pmu_io_domains {
@@ -1356,19 +1333,6 @@ vcc_4g_5v_en: vcc-4g-5v-en-pin {
 		vcc_4g_en: vcc-4g-en-pin {
 			rockchip,pins = <4 RK_PC7 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
-
-		modem_control_pins: modem-control-pins {
-			rockchip,pins =
-				<0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PB0 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PA6 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PB5 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PB4 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PA1 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PA3 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PC1 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
 	};
 
 	pmic {
-- 
2.44.0

